priv/perf/apps/load_test/lib/child_chain/child_chain.ex

```diff
@@ -0,0 +1,89 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+defmodule LoadTest.ChildChain.Transaction do
+  @moduledoc """
+  Utility functions for sending transaction to child chain
+  """
+  require Logger
+
+  alias ExPlasma.Encoding
+  alias ExPlasma.Transaction
+
+  @retry_interval 1_000
+
+  def submit_tx(inputs, outputs, signers, retries \\ 0) do
+    {:ok, tx} = Transaction.Payment.new(%{inputs: inputs, outputs: outputs})
+
+    keys =
+      signers
+      |> Enum.map(&Map.get(&1, :priv))
+      |> Enum.map(&Encoding.to_hex/1)
+
+    tx
+    |> Transaction.sign(keys: keys)
+    |> try_submit_tx(retries)
+  end
+
+  defp try_submit_tx(tx, 0), do: do_submit_tx(tx)
+
+  defp try_submit_tx(tx, retries) do
+    case do_try_submit_tx(tx) do
+      :retry ->
+        Process.sleep(@retry_interval)
+        try_submit_tx(tx, retries - 1)
+
+      result ->
+        result
+    end
+  end
+
+  defp do_try_submit_tx(tx) do
+    case do_submit_tx(tx) do
```

Comment:
is it important that you have so many cases here for handling these results?
bottom up:
```
%{"code" => reason} ->
        _ = Logger.warn("Transaction submission has failed, reason: #{inspect(reason)}")
        {:error, reason}
```
```
{:error, "submit:utxo_not_found"} -> :retry
```
```
:retry ->
```

wouldn't it be simpler if you just did that once, here: https://github.com/omisego/elixir-omg/pull/1403/files#diff-4f5dd40531f9c00595c4790bf859b585R41

Reply:
```
case do_try_submit_tx(tx) do
      {:error, "submit:utxo_not_found"} ->
        Process.sleep(@retry_interval)
        try_submit_tx(tx, retries - 1)

      result ->
        result
    end
```
and you can remove `do_try_submit_tx/1`

Reply:
Yes, that was bogus.